spec/spec_helper.rb

```diff
@@ -121,4 +121,12 @@
   # test failures related to randomization by passing the same `--seed` value
   # as the one that triggered the failure.
   Kernel.srand config.seed
+
+  # Explicitly set a seed to ensure deterministic Percy snapshots.
+  config.around(:each, js: true) do |example|
+    prev_random_seed = Faker::Config.random
+    Faker::Config.random = Random.new(42)
+    example.run
+    Faker::Config.random = prev_random_seed
+  end
```

Comment:
```suggestion
    config.around(:each, js: true) do |example|
    prev_random_seed = Faker::Config.random
    Faker::Config.random = Random.new(42)
    Timecop.freeze("2019-03-04T10:00:00Z")

    example.run

    Faker::Config.random = prev_random_seed
    Timecop.return
  end
```

Reply:
@rhymes Yeah, I know what you mean. The other option that I can think of is something like this:
```ruby
# rails_helper.rb

config.around(:each, freeze_time: true) do |example|
  Timecop.freeze("2019-03-04T10:00:00Z")
  example.run
  Timecop.return
end
```

and then wrapping the Percy examples that specifically need a deterministic time set like this:
```ruby
it "renders the page", js: true, freeze_time: true do
  Percy.snapshot(page, name: "User: /:user_id for logged in user's own profile")
end
```

I don't love it, but it is the more conservative course of "change only what we need to change" rather than freezing time for all the JS tests (which is breaking things anyways!).

Reply:
> Can you give more details on which examples failed with that error?

@rhymes I pushed my commit so now you can see my many, many failing specs that show up when I try to freeze time for those `js: true` specs :(

Reply:
I'll take a look, thanks!

Reply:
I honestly have no idea why they fail, the first thing i can think of is the frozen time being too much in the past, this tricks the Chrome process which thinks is 2019 and then it fails? I wonder what those local capybara snapshots contain. Another thing I could try to check if what happens if you set that datetime with `date` in the terminal and then start Chrome manually